src/cursor/abstract_cursor.ts

```diff
@@ -209,6 +209,17 @@ export abstract class AbstractCursor<
     }
   }
 
+  /**
+   * The cursor has no id until it receives a response from the initial cursor creating command.
+   *
+   * It is non-zero for as long as the database has an open cursor.
+   *
+   * The initiating command may receive a zero id if the entire result is in the `firstBatch`.
+   *
+   * The id value may still be non-zero after calling `cursor.close()`
```

Comment:
This is another regression - we previously zero'ed out the cursor id when closing it:

```
if (cursorId == null || server == null || cursorId.isZero() || cursorNs == null) {
    if (needsToEmitClosed) {
      cursor[kClosed] = true;
      cursor[kId] = Long.ZERO;  <-- zero out id
      cursor.emit(AbstractCursor.CLOSE);
    }

    if (session) {
      if (session.owner === cursor) {
        await session.endSession({ error });
        return;
      }

      if (!session.inTransaction()) {
        maybeClearPinnedConnection(session, { error });
      }
    }

    return;
  }

```

Reply:
Good catch, fixed!